tjreports/administrator/controllers/indexer.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * @package     TJReports
+ * @subpackage  com_tjreports
+ *
+ * @author      Techjoomla <extensions@techjoomla.com>
+ * @copyright   Copyright (C) 2009 - 2019 Techjoomla. All rights reserved.
+ * @license     http://www.gnu.org/licenses/gpl-2.0.html GNU/GPL
+ */
+
+// No direct access.
+defined('_JEXEC') or die('Restricted access');
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\MVC\Controller\BaseController;
+
+/**
+ * TJReports Indexer Controller Class
+ *
+ * @package     TJReports
+ * @subpackage  com_tjreports
+ *
+ * @since       1.1.0
+ */
+class TjreportsControllerIndexer extends BaseController
+{
+	/**
+	 * Function to get all the respective plugins for given client
+	 *
+	 * @return  object  object
+	 */
+	public function createTable()
```

Comment:
Should we whitelist currently supported contexts to prevent erroneous creation of tables ?

Reply:
done - https://github.com/techjoomla/com_tjreports/pull/128/files#diff-0c3d739be62b4e1ba5c1277fbdcf42eaR32